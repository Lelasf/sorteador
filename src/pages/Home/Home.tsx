import Card from "../../components/Card/Card";
import Footer from "../../components/Footer/Footer";
import Form from "../../components/Form/Form";
import ParticipantsList from "../../components/EntrantsList /ParticipantsList";

const Home = () => {
  return (
    <Card>
      <section>
        <h2>Vamos Come√ßar!</h2>
        <Form />
        <ParticipantsList />
        <Footer />
      </section>
    </Card>
  );
};

export default Home;
