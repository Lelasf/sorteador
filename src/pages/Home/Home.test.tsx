import { RecoilRoot } from "recoil";
import Home from "./Home";
import { render } from "@testing-library/react";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => {
  const actualModule = jest.requireActual("react-router-dom");
  return {
    __esModule: true,
    ...actualModule,
    useNavigate: () => mockNavigate,
  };
});

describe("a pagina de configuração", () => {
  test("deve ser renderizada corretament", () => {
    const { container } = render(
      <RecoilRoot>
        <Home />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
