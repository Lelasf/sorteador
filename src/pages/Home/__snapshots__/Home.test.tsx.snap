// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuração deve ser renderizada corretament 1`] = `
<div>
  <div
    class="sc-FEMpB gAXQMB"
  >
    <section>
      <h2>
        Vamos Começar!
      </h2>
      <form>
        <div
          class="sc-dycHyt HIeeK"
        >
          <input
            class="sc-jbAkgO eTOYl"
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            class="sc-gNZgCX cIhVaL"
            disabled=""
          >
            Adcionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="sc-gjcoXW jOqdtr"
      >
        <button
          class="sc-dVBluf edbLxr"
          disabled=""
        >
          Iniciar Brincadeira!
        </button>
        <img
          alt="Sacolas de compras"
          src="/images/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
